### PR TITLE
Add SQLDataSources function

### PR DESCRIPTION
**Pull request by mpcjanssen**

### CODE REVIEW

**Comment by mpcjanssen** [comment]
Apologies for the renamed pkg path. I would love to find a good way to handle this which doesn't require all kinds of tricks on the development side.

**Comment by alexbrainman** [comment]
> Apologies for the renamed pkg path.

You cannot rename pkg path, because this package users refer to this package as github.com/alexbrainman/odbc in their code. So Go compiler will not see your changes.

> I would love to find a good way to handle this which doesn't require all kinds of tricks on the development side.

git provides info tools to overcome this problem. Maybe read https://splice.com/blog/contributing-open-source-git-repositories-go/ - it should help you to make your git copy work with original import paths. Let me know, if you still confused and need more help.

Alex

File: env.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/mpcjanssen/odbc/api"
+	"unsafe"
+)
+
+type DSN struct {
+	Name string
+	Description string
+}
+
+func DataSources() ([]DSN, error) {
```
Comment (alexbrainman) [inline env.go]: This needs documentation explaining what it does. Same for DSN.

And commit description needs to explain what is added and why it is useful.

And I am not sure that it is useful. So, please, discuss what you are trying to achieved before investing more time into this.

Comment (mpcjanssen) [reply env.go]: I am building a REST-ish proxy to an ODBC database. In order to test the proxy I made a small web form frontend to execute queries. 
The point of this addition is to make it possible to select a DSN from a dropdown box.
I currently don't have the time to go through the pkgname rewriting link you posted (thanks for that though) so for now I will keep this functionality in my own fork.

Comment (alexbrainman) [reply env.go]: > I currently don't have the time to go through the pkgname rewriting link you posted (thanks for that though) so for now I will keep this functionality in my own fork.

Fair enough. SGTM.

Alex

**Pull request closed without merging**